File: app/src/main/java/com/example/livedatademo/MainActivity.java
package com.example.livedatademo;

import android.os.Bundle;
import android.view.View;

import com.example.livedatademo.ui.main.MainFragment;
import com.example.livedatademo.ui.repos.ReposFragment;

import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.main_activity);
        if (savedInstanceState == null) {
            getSupportFragmentManager().beginTransaction()
                    .replace(R.id.container, MainFragment.newInstance())
                    .commitNow();
        }
    }


    public void showRepos(View view) {
        getSupportFragmentManager().beginTransaction()
                .replace(R.id.container, ReposFragment.newInstance())
                .commitNow();
    }
}

File: app/src/main/java/com/example/livedatademo/MainActivity.kt
package com.example.livedatademo

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.navigation.fragment.NavHostFragment

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.main_activity)
        val finalHost = NavHostFragment.create(R.navigation.nav_graph)
        supportFragmentManager.beginTransaction()
                .replace(R.id.nav_host, finalHost)
                .setPrimaryNavigationFragment(finalHost) // this is the equivalent to app:defaultNavHost="true"
                .commit()
    }
}
